### PR TITLE
fix: fixes right body type not selected after curl import

### LINKED ISSUE #13978: [Bug]: Curl Import Not Detecting Right Body Type
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

Whenever a URL is imported via CURL, the right body type is not detected and always defaults to RAW

### Steps To Reproduce

Try importing this multipart form request

```curl
curl --location --request POST 'https://api.stripe.com/v1/transfers' \
--header 'Authorization: Basic c2tfdGVzdF81MUpqSlh0RlRiN09mV3VhUjFTcXJoNFF3cU5haDZHcDBhanRpMGhqZlczWjl4MXg4RzFRYUhxNlRZcXBvd1IwVGczcDFadWViQjAwSHB4VnBiUmg6' \
--header 'Content-Type: application/x-www-form-urlencoded' \
--data-urlencode 'amount=1' \
--data-urlencode 'currency=usd' \
--data-urlencode 'destination=acct_1L0mdx2Ym60mujGg'
```

### Public Sample App

_No response_

### Version

Cloud/v1.6.26-SNAPSHOT

### PR DESCRIPTION
**Pull request by sneha122**
## Description

When URL is imported via CURL, right body type is not selected and always defaults to RAW, this PR fixes that issue so that when user clicks on body tab, it defaults to right body format.

Fixes #13978 

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

- Manually Tested
- Jest Unit test

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by AmanAgarwal041 — DISMISSED** [review]

File: app/server/appsmith-server/src/test/java/com/appsmith/server/services/CurlImporterServiceTest.java [hunk 1]
```diff
@@ -848,6 +851,24 @@ public void importInvalidCurlCommand() {
                 .verifyError();
     }
 
+    @Test
+    public void checkActionConfigurationFormData() {
```
Comment (sumitsum) [inline app/server/appsmith-server/src/test/java/com/appsmith/server/services/CurlImporterServiceTest.java:855]: Can you please rename this to `checkActionConfigurationFormDataForApiContentKey` ?

File: app/server/appsmith-server/src/main/java/com/appsmith/server/services/ce/CurlImporterServiceCEImpl.java [hunk 2]
```diff
@@ -50,6 +51,7 @@ public class CurlImporterServiceCEImpl extends BaseApiImporter implements CurlIm
     private static final String ARG_COOKIE = "--cookie";
     private static final String ARG_USER = "--user";
     private static final String ARG_USER_AGENT = "--user-agent";
+    private static final String FORM_DATA_KEY = "apiContentType";
```
[the remark this answers is not in the document]
Comment (sondermanish) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/services/ce/CurlImporterServiceCEImpl.java]: modified as suggested

File: app/server/appsmith-server/src/main/java/com/appsmith/server/services/ce/CurlImporterServiceCEImpl.java [hunk 3]
```diff
@@ -347,9 +349,14 @@ public ActionDTO parse(List<String> tokens) throws AppsmithException {
                 }
                 if ("content-type".equalsIgnoreCase(parts[0])) {
                     contentType = parts[1];
+                    parts[0] = HttpHeaders.CONTENT_TYPE;
```
[the remark this answers is not in the document]
Comment (sondermanish) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/services/ce/CurlImporterServiceCEImpl.java:353]: modified as suggested.

File: app/server/appsmith-server/src/test/java/com/appsmith/server/services/CurlImporterServiceTest.java (hunk 1 above)
Comment (sondermanish) [reply app/server/appsmith-server/src/test/java/com/appsmith/server/services/CurlImporterServiceTest.java]: modified as suggested.

**Comment by sondermanish** [comment]
\ok-to-test sha=9082a27

**Review by sumitsum — DISMISSED** [review]

**Review by AmanAgarwal041 — DISMISSED** [review]

**Review by AmanAgarwal041 — APPROVED** [review]

**Pull request merged**